A/AOCL_BLAS/common.jl

```diff
@@ -0,0 +1,107 @@
+# Note that this script can accept some limited command-line arguments, run
+# `julia build_tarballs.jl --help` to see a usage message.
+using BinaryBuilder, Pkg
+
+version = v"5.1.0"
+
+# Collection of sources required to complete build
+sources = [
+    GitSource("https://github.com/amd/blis.git", "16f852a065e76e824d77bc39e2baa82ac19ed419"),
+    DirectorySource("../bundled")
+]
+
+# Bash recipe for building across all platforms
+# Most of the script is adapted from blis
+function blis_script(; blis32::Bool=false)
+    script = """
+    BLIS32=$(blis32)
+    """
+
+    script *= raw"""
+    cd $WORKSPACE/srcdir/blis
+
+    for i in ./config/*/*.mk; do
+
+        # Building in container forbids -march options <<< Settings overriden.
+        # sed -i "s/-march[^ ]*//g" $i
+
+        # Building in container forbids unsafe optimization.
+        sed -i "s/-ffast-math//g" $i
+        sed -i "s/-funsafe-math-optimizations//g" $i
+
+    done
+
+    # Only consider AMD Zen CPUs
+    export BLI_CONFIG=amdzen
+    export BLI_THREAD=openmp
+
+    # For 64-bit builds, add _64 suffix to exported BLAS routines.
+    # This corresponds to ILP64 handling of OpenBLAS thus Julia.
+    # if [[ ${nbits} == 64 ]] && [[ "${BLIS32}" != "true" ]]; then
+        # atomic_patch -p1 ${WORKSPACE}/srcdir/patches/suffix64.patch
+    # fi
+
+    # Fix the unnecessary branching for Windows that caused errors/warnings
+    if [[ "${target}" == *"x86_64"*"w64"* ]]; then
+        atomic_patch -p1 ${WORKSPACE}/srcdir/patches/aocltpdef-mingw32.patch
+        atomic_patch -p1 ${WORKSPACE}/srcdir/patches/blis_tls_type-mingw32.patch
+    fi
+
+    # Import libblastrampoline-style nthreads setter.
+    cp ${WORKSPACE}/srcdir/nthreads64_.c frame/compat/nthreads64_.c
+
+    if [[ "${BLIS32}" == "true" ]]; then
+        export BLI_F77BITS=32
+    else
+        export BLI_F77BITS=${nbits}
+    fi
+    ./configure --enable-cblas -p ${prefix} -t ${BLI_THREAD} -b ${BLI_F77BITS} --enable-aocl-dynamic ${BLI_CONFIG}
+    make -j${nproc}
+    make install
+
+    # Static library is not needed
+    rm ${prefix}/lib/libblis-mt.a
+
+    # Rename .dll for Windows targets.
+    if [[ "${target}" == *"x86_64"*"w64"* ]]; then
+        mkdir -p ${libdir}
+        mv ${prefix}/lib/libblis-mt.*.dll ${libdir}/libblis-mt.dll
+    fi
```

Comment:
Why do you need to rename the files?

Reply:
@giordano I followed what was done for blis_jll here: https://github.com/JuliaPackaging/Yggdrasil/blob/master/B/blis/common.jl I don't know why they did some renaming towards the end, but assume they did this for good reasons? If you think this is uncessary, then these lines could just be removed.

Reply:
@giordano From my local run, I see the file moved to `libdir` is libblis-mt.5.dll. I guess the file is renamed simply because the digit `5` is something that could change?

Reply:
The BLIS recipe needs a bit of modernization, so not everything it is doing is needed in this script. We don't really have a reason to do any renaming of the library, as long as the `LibraryProduct` is given the proper names to search for.

I think keeping the 5 is better here also, because that gives information about the library version (it is major version 5).